pallets/crowdloan-claim/src/lib.rs

```diff
@@ -810,6 +830,11 @@ impl<T: Config> Pallet<T> {
 			Error::<T>::InvalidProofOfContribution
 		);
 
+		ensure!(
```

Comment:
Yes, the leaf always has to be generated or checked based on the raw data, thank you for catching it.